src/vs/workbench/contrib/chat/browser/actions/chatToolPicker.ts

```diff
@@ -412,6 +419,10 @@ export async function showToolsPicker(
 			if (!tool.canBeReferencedInPrompt || !toolsEntries.has(tool)) {
```

Comment:
this flag `canBeReferencedInPrompt` is causing tools with this flag to be shown at top-level and in the toolset, not sure if we still need this block at all? @connor4312